unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/BoxShape/Handler/ECSBoxShapeComponentHandler.cs

```diff
@@ -58,12 +62,22 @@ public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBBox
 
         private void GenerateRenderer(Mesh mesh, IParcelScene scene, IDCLEntity entity, PBBoxShape model)
         {
-            meshesInfo = ECSComponentsUtils.GenerateMeshInfo(entity, mesh, entity.gameObject, model.Visible, model.WithCollisions, model.IsPointerBlocker);
+            GenerateHolder(entity);
+            meshesInfo = ECSComponentsUtils.GenerateMeshInfo(entity, mesh, meshHolderGameObject, model.Visible, model.WithCollisions, model.IsPointerBlocker);
 
             // Note: We should add the rendereable to the data store and dispose when it not longer exists
             rendereable = ECSComponentsUtils.AddRendereableToDataStore(scene.sceneData.id, entity.entityId, mesh, entity.gameObject, meshesInfo.renderers);
         }
 
+        private void GenerateHolder(IDCLEntity entity)
+        {
+            if(meshHolderGameObject != null)
+                GameObject.Destroy(meshHolderGameObject);
+            
+            meshHolderGameObject = new GameObject();
+            meshHolderGameObject.transform.SetParent(entity.gameObject.transform,false);
+        }
```

Comment:
I'm seeing this `GenerateHolder` function is being repeated in several classes... maybe we could extract it to some kind of common/utility class?

Reply:
I have removed it!